airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/SettingsView.tsx

```diff
@@ -15,11 +16,12 @@ interface SettingsViewProps {
 const SettingsView: React.FC<SettingsViewProps> = ({ connection }) => {
   const { mutateAsync: deleteConnection } = useDeleteConnection();
 
+  const [isAdvancedMode] = useAdvancedModeSetting();
```

Comment:
It doesn't look like there is a way for users to turn this mode on in OSS.  I only see the Cloud settings page.  Is that intentional?

Reply:
I wondered about that, too! I couldn't find any workspace settings page outside of cloud, though:
```
┌ ~/c/airbyte/airbyte-webapp (amb/add-advanced-mode-toggle|✔) 🌗 a3a86ce75c
└➣ fd workspace | egrep -v '^\./src/packages/cloud/' | egrep '\.[^/]+' # filter out cloud-specific results and directories
./src/core/domain/workspace/Workspace.ts
./src/core/domain/workspace/WorkspaceService.ts
./src/hooks/services/useWorkspace.tsx
./src/pages/SettingsPage/components/useWorkspaceEditor.tsx
./src/services/workspaces/WorkspacesService.tsx
```

When I looked through those files and the places that used them, I couldn't find any that seemed equivalent to the cloud workspace settings page (`src/views/Settings/PreferencesForm/PreferencesForm.tsx` seemed to be the closest, but the other fields were concerned with updates and telemetry, rather than product feature configuration).

I'll fire up a local backend and do a bit more investigation, though, thanks for calling that out.

Reply:
Yeah.  OSS doesn't use workspaces.  Hmm.  I remember chatting about this with @edmundito and @timroes at one point and I think we settled on this being a new settings page for OSS? (like Account, Sources, Destinations etc. are)

Reply:
Technically OSS deployments do use a single workspace under the hood (and I believe users can also use the API to create more workspaces, and switch between them by changing the workspace ID in the URL), so if we are also applying this change to OSS we will still want to key the local storage values based on workspace ID like you are doing in this PR.

But yeah, we will probably need a new settings page to contain this toggle.

Reply:
Cool, thanks for clarifying!

Reply:
Come to think of it, I think it would make some sense to put this within "Account Settings" for OSS, at least so long as the OSS version uses a singleton workspace. My thinking is that each advanced mode setting is logically scoped to a specific account/workspace pairing; the user account is implicit (settings are always changed for the logged-in user)*, so for the cloud version of the app it makes sense to put it under workspace settings. For OSS, the workspace is implicit, too (only one workspace per OSS instance), but the logical dependency on a user account is still there, so it feels natural enough to me for the OSS "Advanced Mode" setting to be in the account settings page--at least, it feels less arbitrary than a whole new "Advanced Mode"/"Viewing settings"/etc settings page.

Reply:
What you said makes sense. The one counterargument I have is that _if_ some OSS user has created multiple workspaces using the API and they commonly switch between them by changing the workspace ID in the URL, then it may be slightly confusing that there is a toggle affecting specific workspaces that lives in `Account Settings`, as to me Account Settings implies that these are configurations for my entire account, not an individual workspace.

However, I don't think that this is the typical way OSS users use the app - I think typically they use it as you describe, where they aren't even really aware that they are using a "workspace" and they will only ever have a single account and workspace.

Though, I could see us eventually wanting to make workspaces more of a first-class feature in the OSS app (e.g. providing a UI to create and switch workspaces similar to Cloud). Not sure what the timeline of that would be but I'm pretty sure I've heard that it is a requested feature.

All of that said, I don't have super strong feelings on this and I think either option would be fine, both with their slight tradeoffs. So I'm happy with deferring to what you think is best here!